DataBiosphere/terra-cli
# [PF-2174] check version is out of date for the commands talk to server

This PR comes from the bug ticket [PF-2174](https://broadworkbench.atlassian.net/browse/PF-2174)

Before the ticket, we do the version check every 30 minutes and do not check the version for every command. 
Because we do not support the version out of date. We should throw error to user when they request the server but the version is old.

The ticket scope:
- create the subclass which extend from the parent class `BaseCommand`. and can specify part of the command use the subclass. and rest of them still use the parent class.
- only affect the resource commands and workspace commands because they talk to the server.
- remove the existing expire time check in `src/main/java/bio/terra/cli/app/utils/VersionCheckUtils.java` 

Manually test:

the version is up to date:
```
ginay-macbookpro:terra-cli ginay$ terra workspace list
   ID                                        NAME                            GOOGLE PROJECT                  DESCRIPTION                             
   propertytest                              (unset)                         terra-vdevel-halcyon-onion-400  (unset)                                 
   junhtest                                  (unset)                         terra-vdevel-smart-mango-3426   (unset)                                 
   a-6bc4a502-8600-4449-b224-0104d80e444b    (unset)                         terra-vdevel-speedy-lemon-8435  (unset)                                 
   clonetestname                             (unset)                         terra-vdevel-cold-lime-8223     (unset)                                 
   newidfortestnameclone                     (unset)                         terra-vdevel-happy-coffee-83    (unset)                                 
   newidupdated                              (unset)                         terra-vdevel-genial-grape-9381  (unset)      
```

manually set the version to old version:
```
ginay-macbookpro:terra-cli ginay$ terra workspace list
Warning: Version 0.27.0 of the CLI has expired. Functionality may not work as expected. To install the latest version: curl -L https://github.com/DataBiosphere/terra-cli/releases/latest/download/download-install.sh | bash ./terra
If you have added the CLI to your $PATH, this step will need to be repeated after the installation is complete.
```
It does not affect on the command which not request the server
```
ginay-macbookpro:terra-cli ginay$ terra config list
OPTION                VALUE                                          DESCRIPTION                                                 
app-launch            DOCKER_CONTAINER                               app launch mode                                             
browser               AUTO                                           browser launch for login                                    
image                 gcr.io/terra-cli-dev/terra-cli/0.2.0:stable    docker image id                                             
resource-limit        1000                                           max number of resources to allow per workspace              
console-logging       OFF                                            logging level for printing directly to the terminal         
file-logging          INFO                                           logging level for writing to files/Users/ginay/.terra/logs  
server                verily-devel                                   (unset)                                                     
workspace             testid                                         (unset)                                                     
format                TEXT                                           output format       
```
